### DISCUSSION TITLE
New big change if possible

### DISCUSSION THREAD

**Discussion by AndrejWeber** [theyosh/TerrariumPI#757] [Ideas]
Hey Yosh,
I am using your terrariumPi software since a few years (you wrote the DC-dimmer for me), and I am still active in reading the mails, coming by GitHub. 
The raspi incl terrarium is not at my home since a few years because my son went out, so I am not active here with new features and so on. 
I do administrate this terrarium once or two time a year, and it works great. 
But today I had an idea, what would be very useful, but not sure, it is possible at all. 
Most problems with terrariumPi appear, when the user is changing or trying anything. The the startup routine will crash and you try it about 10 times, then the service stops.
A new idea would be, instead of stopping the service, you would start a backup config, without any sensor and nothing else, just a working GUI. 
The next thing we need, if there would be a possibility, to change the config file in the GUI. (Therefor you have to add a text editor in the GUI, I do not now, if it is possible at all.) After changing the config file, the user can restart the service to load the original config file.
It also could be useful, when every little change will get a time stamp in the config file.
So if you change the address or pin of a sensor, you will find easily the right peace of config file, you have to edit, or even to delete.
May be, it is too much rework and hard to implement, but it could be something you can add in the next big version change. 
Thank you for all this great work!

**Comment by theyosh** [comment]
He Andrej,

sorry for the late reply. I have thought about this, but with version 4, all settings are in the database. So it is a bit hard to make a backup of that every time you make a change. You only need the settings, but that means dumping almost hole database every time you update something.

Also, the restarts are outside TP4. That is done by systemd, a process of the OS. So I do not know that a restart has occurred. And also not the reason.

My solution is to backup the database every night. And keep about 10 copies. So when something goes wrong, I can go back 1 day. You will lose some sensor and relay history, but that is the easiest way to recover from a wrong config, which could not be found.

So, I get the feeling you are still on version 3? That is not supported anymore....
